enterprise/frontend/src/embedding-sdk/store/auth.ts

```diff
@@ -0,0 +1,158 @@
+        "Failed to fetch the site settings, is the session valid?",
+      );
+      console.error(error);
+      throw error;
+    }
+  },
+);
+
+export const refreshTokenAsync = createAsyncThunk(
+  "sdk/token/REFRESH_TOKEN",
+  async (url: string, { getState }): Promise<EmbeddingSessionToken | null> => {
+    // The SDK user can provide a custom function to refresh the token.
+    const customGetRefreshToken = getFetchRefreshTokenFn(
+      getState() as SdkStoreState,
+    );
+
+    const getRefreshToken = customGetRefreshToken ?? defaultGetRefreshTokenFn;
+
+    // # How does the error handling work?
+    // This is an async thunk, thunks _by design_ can fail and no error will be shown on the console (it's the reducer that should handle the reject action)
+    // The following lines are wrapped in a try/catch block that will catch any errors thrown, log them to the console as a big red errors, and re-throw them to make the thunk reject
+    // In this way we also support standard thrown Errors in the custom fetchRequestToken user provided function
+
+    try {
+      const response = await getRefreshToken(url);
+      const source = customGetRefreshToken
+        ? '"fetchRequestToken"'
+        : "jwtProvider endpoint";
+
+      if (!response || typeof response !== "object") {
+        throw new Error(
+          `The ${source} must return an object with the shape {id:string, exp:number, iat:number, status:string}, got ${safeStringify(response)} instead`,
+        );
+      }
+      if ("status" in response && response.status !== "ok") {
+        if ("message" in response && typeof response.message === "string") {
+          // For some errors, the BE gives us a message that explains it
+          throw new Error(response.message);
+        }
+        if ("status" in response && typeof response.status === "string") {
+          // other times it just returns an error code
+          throw new Error(
+            `Failed to refresh token, got status: ${response.status}`,
+          );
+        }
+      }
+      // Lastly if we don't have an error message or status check if we actually got the session id
+      if (!("id" in response)) {
+        throw new Error(
+          `${source} must return an object with the shape {id:string, exp:number, iat:number, status:string}, got ${safeStringify(response)} instead`,
+        );
+      }
+      return response;
+    } catch (ex: unknown) {
+      if (ex instanceof Error) {
+        Error.captureStackTrace(ex, refreshTokenAsync);
+      }
+
+      // The host app may have a lot of logs (and the sdk logs a lot too), so we
+      // make a big red error message to make it visible as this is 90% a blocking error
+      console.error(
+        "%cFailed to get auth session\n",
+        "color: #FF2222; font-size: 16px; font-weight: bold;",
+        ex,
+      );
+
+      throw ex;
+    }
+  },
+);
+
+const safeStringify = (value: unknown) => {
+  try {
+    return JSON.stringify(value);
```

Comment:
if we want the output to be a bit more friendly, should we indent it?

```suggestion
    return JSON.stringify(value, null, 2);
```

Reply:
I would expect only small objects/json to be returned from the endpoint, and they would look a bit weird in the middle of the log if they're indented:
<img width="1103" alt="image" src="https://github.com/user-attachments/assets/85244445-ef67-47e5-9d4c-205e3de58f59">
We could check if it's a json and add some new lines but I'm not sure it's worth it, as long as they understand what's being returned (ie: a reverse proxy error, a string etc) it should be fine, they can inspect the network tab to see the details

Reply:
that's a good point, thanks for including the screenshot. Feels free to leave it as is.